forge-game/src/main/java/forge/game/ability/effects/HeistEffect.java

```diff
@@ -0,0 +1,65 @@
+package forge.game.ability.effects;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Predicates;
+
+import forge.game.Game;
+import forge.game.ability.AbilityKey;
+import forge.game.ability.AbilityUtils;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.card.CardCollection;
+import forge.game.card.CardLists;
+import forge.game.card.CardPredicates;
+import forge.game.card.CardZoneTable;
+import forge.game.player.Player;
+import forge.game.spellability.SpellAbility;
+import forge.game.zone.ZoneType;
+import forge.util.Aggregates;
+import forge.util.Localizer;
+
+public class HeistEffect extends SpellAbilityEffect {
+
+    @Override
+    public void resolve(SpellAbility sa) {
+        Map<AbilityKey, Object> moveParams = AbilityKey.newMap();
+        moveParams.put(AbilityKey.LastStateBattlefield, sa.getLastStateBattlefield());
+        moveParams.put(AbilityKey.LastStateGraveyard, sa.getLastStateGraveyard());
+        final Card source = sa.getHostCard();
+        final Player player = AbilityUtils.getDefinedPlayers(source, sa.getParam("Defined"), sa).get(0);
+        final Game game = player.getGame();
+        final Player target = getTargetPlayers(sa).get(0);
+        final int num = AbilityUtils.calculateAmount(source, sa.getParamOrDefault("Num", "1"), sa);
+        final CardZoneTable triggerList = new CardZoneTable();
+
+        for (int i = 0; i < num; i++) {
+            List<Card> list = CardLists.filter(target.getCardsIn(ZoneType.Library),
```

Comment:
use .getNotType

Reply:
This combined should do the trick:

`List<Card> choices = Aggregates.random(CardLists.getNotType(target.getCardsIn(ZoneType.Library), "Land"), 3)`